app/gui2/src/components/GraphEditor/widgets/WidgetPort.vue

```diff
@@ -136,6 +136,7 @@ export const widgetDefinition = defineWidget(
       Ast.Tree.Type.UnaryOprApp,
       Ast.Tree.Type.Wildcard,
       Ast.Tree.Type.TextLiteral,
+      Ast.Tree.Type.Ident,
```

Comment:
oh... i can't think of a way to make this work. do you have any ideas?

Reply:
it seems like the problem is: we need to bail out of selecting *just `WidgetPort`s*, but not other widget types because we still need to render the actual AST subtree?

i was thinking that maybe we'd need to be able to un-bail-out of selecting `WidgetPort`s?
on one hand, *theoretically* they shouldn't be valid ASTs...
on the other hand, i'm assuming they're entirely possible to have, if they were already present when the file was originally opened.

Reply:
it'd always be an option to create an extra class to mark something as needing to bypass `WidgetPort`, like `ForcePort` already does, but I'm not sure that's an ideal option...

it'd also be an option to be able to attach arbitrary metadata to an `AstExtended`... that for some reason should get passed to children. i don't imagine it'd be too bad implementation-wise... but it sounds like a headache to conceptualize.